hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/CompactStreamSerializer.java

```diff
@@ -323,4 +322,31 @@ private void registerConfiguredNamedSerializers(CompactSerializationConfig compa
             typeNameToRegistrationMap.put(typeName, registration);
         }
     }
+
+    public Schema extractSchema(BufferObjectDataInput objectDataInput) throws IOException {
+        return getOrReadSchema(objectDataInput, false);
+    }
+
+    public Schema extractSchema(Object o) {
+        CompactSerializableRegistration registration = getOrCreateRegistration(o);
```

Comment:
Would it make sense to get or create registration only if the schema we are looking for is not in the `classToSchemaMap`? (basically in the if block, where we need it)